CONTRIBUTING.md

```diff
@@ -210,6 +214,231 @@ it is helpful to use [npm link](https://docs.npmjs.com/cli/link).
     }
     ```
 
+## Testing
+
+### Why is testing necessary? 
+
+Many of us like to sleep at night. So to give us peace of mind when we release a
+new version of our software, we want to guarantee that the application works as 
+it should, always. To accomplish this, we write two kinds of tests that will 
+ensure that our applications behave as it should even as we add new features.
+
+### Organizing Tests
+
+It is common practice, to group automated tests in categories depending on how 
+broad their scope is, and how complex their execution requirements are. 
+This separation often follows the 
+[testing pyramid](https://martinfowler.com/bliki/TestPyramid.html). 
+In DC/OS UI we use three levels of testing: Unit, Integration, and System.
+
+[Unit Tests](#unit-testing) are intended to verify functionality at the minimal 
+level of isolation, for that purpose, they tend to be small, and mock/stub all 
+its dependencies. They also tend to execute fast and in a reliable way.
+
+Given a system is composed of a collection of components, that when tested 
+in isolation perform correctly, but often fail to do so together, we use both 
+integration testing and system testing to verify behavior at an increasing 
+larger scope (bringing increased execution requirements).
+
+[Integration Tests](#integration-testing) verify that the well-tested units do 
+not fail when they interact with components they depend on internally (part of 
+the same project), but will stub/mock external dependencies (from other 
+systems/projects). Compared to unit testing, integration tests tend to be harder 
+to execute, since they rely on an environment similar to production (e.g. they 
+run ion the browser). They also tend to be slower than unit tests making it 
+impossible to exercise all possible scenarios.
+
+While Integration Tests looks for the composition within the system, 
+[System Tests](#system-testing) test the integration with the real external 
+dependencies trying to as close as possible to the production use of a client 
+(in the case of DC/OS, against a real cluster). They are even harder to setup 
+than integration tests because you often have to create custom tooling to 
+operate the whole product. This contributes to them being slower and more often 
+prone to fail occasionally, for example, because of network issues like latency.
+
+## Unit Testing
```

Comment:
What is the purpose fo this paragraph?

Reply:
Not quite sure which paragraph we are talking about.

Case 1: To ensure that individual...
You can navigate straight to Unit testing via a link, so for a reader that starts at this point of the document, jumping straight to the sum example looked to much, so I provided a small sentence to introduce the topic of unit testing.

Case 2: Think of a sum ...
This was there before, it is an example of unit testing, I left also because I build on this on the later topics (Integration tests).

Reply:
Sorry for being unclear, I will try to improve 👍 

I meant the entire unit testing part, for me it is text that we kind of have to maintain and I think a link might be sufficient.

Reply:
I think it is not a bad idea to have a description about what we do understand by unit tests to set a baseline.

Reply:
🤔 Yeah, I haven't thought about it in terms of setting a baseline.

Reply:
I also like the setting the baseline idea. I am creating a ticket about documentation moving further, so we can keep improving this for us, and our community.

I think also if someone can provide a great tests documentation on the internet, for this basic topics, we could just link and be happy (like the betterspecs.org).